File: src/commands/usage.rs
// Copyright 2026 Hemi Labs, Inc.
// SPDX-License-Identifier: GPL-3.0-only

use anyhow::{Context, Result};

use crate::client::{check_response, ApiClient};
use crate::models::UsageSummary;
use crate::output::{format_time, new_table, OutputMode};

pub async fn run(client: &ApiClient, tenant_id: &str, mode: OutputMode) -> Result<()> {
    let resp = client
        .get(&format!("/v1/tenants/{tenant_id}/usage"))
        .send()
        .await
        .context("Failed to fetch usage")?;
    let resp = check_response(resp).await?;
    let usage: UsageSummary = resp.json().await.context("Failed to parse usage")?;

    match mode {
        OutputMode::Json => {
            let json = serde_json::to_string_pretty(&usage).expect("JSON serialization failed");
            println!("{json}");
        }
        OutputMode::Human => {
            let cents = usage.balance_cents;
            let dollars = f64::from(i32::try_from(cents).unwrap_or(i32::MAX)) / 100.0;
            println!("Credit Balance: ${dollars:.2}\n");

            if usage.recent_transactions.is_empty() {
                println!("No recent transactions.");
            } else {
                let mut table = new_table();
                table.set_header(["DATE", "AMOUNT", "DESCRIPTION"]);
                for tx in &usage.recent_transactions {
                    let amt_dollars =
                        f64::from(i32::try_from(tx.amount_cents).unwrap_or(i32::MAX)) / 100.0;
                    let sign = if tx.tx_type == "credit" { "+" } else { "-" };
                    table.add_row([
                        format_time(&tx.created_at),
                        format!("{sign}${amt_dollars:.2}"),
                        tx.description.clone(),
                    ]);
                }
                println!("{table}");
            }
        }
    }

    Ok(())
}

File: lib/commands/usage.js
// Copyright 2026 Hemi Labs, Inc.
// SPDX-License-Identifier: GPL-3.0-only

import { newTable, formatTime } from "../output.js";

export async function usageCommand(client, tenantId, json) {
  const resp = await client.get(`/v1/tenants/${tenantId}/usage`);
  const usage = await resp.json();

  if (json) {
    console.log(JSON.stringify(usage, null, 2));
    return;
  }

  const dollars = usage.balance_cents / 100;
  console.log(`Credit Balance: $${dollars.toFixed(2)}\n`);

  if (!usage.recent_transactions || usage.recent_transactions.length === 0) {
    console.log("No recent transactions.");
    return;
  }

  const table = newTable();
  table.push(["DATE", "AMOUNT", "DESCRIPTION"]);
  for (const tx of usage.recent_transactions) {
    const amtDollars = tx.amount_cents / 100;
    const sign = tx.tx_type === "credit" ? "+" : "-";
    table.push([
      formatTime(tx.created_at),
      `${sign}$${amtDollars.toFixed(2)}`,
      tx.description,
    ]);
  }
  console.log(table.toString());
}
